File: tests/fixtures/vulnerable_repo/nodejs/vuln.js
// Intentionally vulnerable Node.js fixture for Suzaku Compass tests.
// DO NOT use this code. It exists only as scanner bait.

const _ = require('lodash');
const vm = require('vm');
const child_process = require('child_process');

function unsafeEval(code) {
  return eval(code);
}

function unsafeFunction(code) {
  return new Function(code)();
}

function unsafeVm(code) {
  return vm.runInNewContext(code);
}

function unsafeMerge(target, untrusted) {
  return _.merge(target, untrusted);
}

function unsafeMergeWith(target, untrusted) {
  return _.mergeWith(target, untrusted);
}

function unsafeExec(userInput) {
  child_process.exec('ls ' + userInput);
}

function dynamicRequire(name) {
  return require(name);
}

function unsafeAssign(target, untrusted) {
  Object.assign(target, untrusted);
}

File: tests/fixtures/vulnerable_repo/python/vuln.py
"""Intentionally vulnerable Python fixture for Suzaku Compass tests.

DO NOT use this code. It exists only as scanner bait.
"""

import pickle
import subprocess
import os
import marshal
import xml.etree.ElementTree as ET  # noqa: N817
import yaml


def unsafe_pickle(data: bytes):
    return pickle.loads(data)


def unsafe_yaml(text: str):
    return yaml.load(text)  # missing SafeLoader


def safe_yaml(text: str):
    return yaml.load(text, Loader=yaml.SafeLoader)


def unsafe_subprocess(user_input: str):
    subprocess.run(user_input, shell=True)


def unsafe_os_system(cmd: str):
    os.system(cmd)


def unsafe_eval(expr: str):
    return eval(expr)


def unsafe_exec(code: str):
    exec(code)


def unsafe_marshal(data: bytes):
    return marshal.loads(data)


def unsafe_xml(payload: str):
    return ET.fromstring(payload)
